### PR TITLE
User can view all items up for auction.

### PR DESCRIPTION
**Pull request by xray**

### CODE REVIEW

File: test/renaissance_web/controllers/login_controller_test.exs [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
     assert html_response(conn, 200) =~ "Login"
   end
 
   test "POST /login with correct email and password" do
-    post(build_conn(), "/register", %{"user" => %{email: "mail@mail.com", password: "password"}})
-
     conn =
-      post(build_conn(), "/login", %{"user" => %{email: "mail@mail.com", password: "password"}})
+      build_conn()
+      |> post("/register", %{"user" => @user_params})
```
Comment (kevbuchanan) [inline test/renaissance_web/controllers/login_controller_test.exs]: Rather than relying on using the controllers to put the connection in the expected state (signed in), user a setup block to just create a user and assign to the session, returning a signed in connection.

File: lib/renaissance_web/controllers/auction_controller.ex [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if Auth.signed_in?(conn) do
+      render(conn, "new_auction.html", changeset: changeset)
+    else
+      conn
+      |> redirect(to: Routes.login_path(conn, :login))
+    end
+  end
+
+  def create(conn, params) do
+    changeset =
```
Comment (kevbuchanan) [inline lib/renaissance_web/controllers/auction_controller.ex]: This isn't a changeset.

File: lib/renaissance/auctions/auctions.ex [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      |> format_price()
+
+    Repo.insert(Auction.changeset(%Auction{}, auction))
+  end
+
+  def get_by_title(title) do
+    Repo.get_by(Auction, title: title)
+  end
+
+  def get_all_auctions() do
```
Comment (kevbuchanan) [inline lib/renaissance/auctions/auctions.ex]: Rather than returning a list of an unspecified map here, I'd expect this to return `Auction` struct, with the seller preloaded:

```
Auction
|> preload(:user)
|> Repo.all()
```

This also leads to the mismatch between what you seem to want to call the `:user` association (`:seller`) and what you've called it on the model. In your ecto schema you could change the `:user` association to `:seller`:

`belongs_to :seller, User`

Then in User:

`has_many :auctions, Auction, foreign_key: :seller_id`

File: .gitignore [hunk 4]
```diff
@@ -175,3 +175,73 @@ modules.xml
 /installer/deps
 
 # End of https://www.gitignore.io/api/phoenix
+
+# Created by https://www.gitignore.io/api/macos
```
Comment (kevbuchanan) [inline .gitignore]: Same comment as the other PR (as well as a bunch of others). This file is getting out of hand.

File: lib/renaissance/auctions/auction.ex [hunk 5]
```diff
@@ -18,8 +19,8 @@ defmodule Renaissance.Auction do
   @doc false
   def changeset(user, attrs \\ %{}) do
     user
-    |> cast(attrs, @required_fields ++ @optional_fields)
-    |> validate_required(@required_fields)
+    |> cast(attrs, [:title, :description, :user_id, :price, :end_date])
```
Comment (kevbuchanan) [inline lib/renaissance/auctions/auction.ex]: Why this change? Using the `@required_fields` and `@optional_fields` is the preferred way here. Cuts down on duplication and makes it easier to add new fields.

Comment (xray) [reply lib/renaissance/auctions/auction.ex]: This was an issue with rebasing from [feature-create-auction](https://github.com/xray/Renaissance/tree/feature-create-auction), this will be fixed.

**Comment by xray** [comment]
[[auctions] User can view all items up for auction.](https://trello.com/c/xOCbcYI5/5-auctions-user-can-view-all-items-up-for-auction)

File: lib/renaissance_web/router.ex [hunk 6]
```diff
@@ -16,14 +16,15 @@ defmodule RenaissanceWeb.Router do
...
 
-    get "/", PageController, :index
+    get "/", AuctionController, :index
     get "/register", RegisterController, :new
     post "/register", RegisterController, :register
     get "/login", LoginController, :login
     post "/login", LoginController, :verify
-    get "/auctions", AuctionController, :list
+
+    get "/auctions", AuctionController, :index
```
Comment (ssunday) [inline lib/renaissance_web/router.ex]: You can use resources for this now, right

File: priv/repo/migrations/20190404190924_create_users.exs [hunk 7]
```diff
@@ -3,8 +3,8 @@ defmodule Renaissance.Repo.Migrations.CreateUsers do
 
   def change do
     create table(:users) do
-      add :email, :string
-      add :password_hash, :string
+      add :email, :string, null: false
```
Comment (ssunday) [inline priv/repo/migrations/20190404190924_create_users.exs:6]: This migration is already checked in, right? You shouldn't modify old migrations instead make new ones. If this had already been deployed, it wouldn't actually take effect

Comment (ssunday) [reply priv/repo/migrations/20190404190924_create_users.exs:6]: Also it doesn't convey the history of the changes

File: test/renaissance_web/controllers/auction_controller_test.exs [hunk 8]
```diff
@@ -20,81 +36,65 @@ defmodule RenaissanceWeb.AuctionControllerTest do
...
-          price: "10.00"
-        }
-      })
+      |> post("/auctions", @auction_one)
 
     assert get_flash(conn, :info) == "Auction Created!"
   end
 
-  test "POST /auctions/new fails to create auction when title is blank" do
+  test "POST /auctions fails to create auction when title is blank" do
```
Comment (ssunday) [inline test/renaissance_web/controllers/auction_controller_test.exs]: Couldn't you do `%{@aucton_one | title: ""} or however to merge the maps?

File: lib/renaissance_web/controllers/auction_controller.ex (hunk 2 above)
Comment (rstein66) [reply lib/renaissance_web/controllers/auction_controller.ex]: Resolved with 159a775e493da7136d042a4f76b690b4d8cb2aea

File: priv/repo/migrations/20190404190924_create_users.exs (hunk 7 above)
Comment (rstein66) [reply priv/repo/migrations/20190404190924_create_users.exs:6]: Too true–will resist this cheat going forward :flushed:

File: test/renaissance_web/controllers/auction_controller_test.exs (hunk 8 above)
Comment (rstein66) [reply test/renaissance_web/controllers/auction_controller_test.exs]: Resolved with 159a775e493da7136d042a4f76b690b4d8cb2aea

File: assets/css/app.css [hunk 9]
```diff
@@ -175,7 +183,7 @@ body {
 }
 
 .auction-detail {
-    margin: 0px;
+    margin: 0;
     font-family: Roboto, sans-serif;
```
Comment (ssunday) [inline assets/css/app.css:187]: This style of saying roboto is diff than above

File: lib/renaissance/auctions/auction.ex [hunk 10]
```diff
@@ -26,24 +26,22 @@ defmodule Renaissance.Auction do
   end
 
   defp validate_price(changeset) do
-    item_price = get_change(changeset, :price)
-    zero_dollars = Money.new(000, :USD)
-    one_dollar = Money.new(100, :USD)
+    {_, amount} = Money.Ecto.Amount.Type.dump(get_change(changeset, :price, 0))
 
-    if Money.compare(item_price || one_dollar, zero_dollars) == 1 do
+    if 0 < amount do
```
Comment (ssunday) [inline lib/renaissance/auctions/auction.ex]: could you reverse this ? `0 < amount` is harder to understand than `amount > 0` IMO

File: lib/renaissance/auctions/auctions.ex [hunk 11]
```diff
@@ -4,9 +4,15 @@ defmodule Renaissance.Auctions do
...
+      if Map.has_key?(params, "auction") do
+        params["auction"]
+      else
+        params
+      end
+
+    details =
+      details
       |> add_user_id(email)
-      |> format_end_date()
```
Comment (elizabethengelman) [inline lib/renaissance/auctions/auctions.ex:8]: Why are we removing the `format_end_date` step here?

File: test/renaissance/auctions/auction_test.exs [hunk 12]
```diff
@@ -4,136 +4,78 @@ defmodule Renaissance.Test.AuctionTest do
   require Ecto.Query
   alias Renaissance.Auction
 
+  @now DateTime.utc_now()
+  @after_one_day %{@now | day: @now.day + 1}
+
+  @valid_params %{
```
Comment (elizabethengelman) [inline test/renaissance/auctions/auction_test.exs:10]: I like this refactoring to reduce duplication.

File: lib/renaissance/auctions/auctions.ex (hunk 11 above)
Comment (rstein66) [reply lib/renaissance/auctions/auctions.ex:8]: The form is now using a Phoenix date time input method and so it is able to understand the input without needing to be told

File: lib/renaissance/auctions/auction.ex (hunk 10 above)
Comment (rstein66) [reply lib/renaissance/auctions/auction.ex]: Sure, will do.
_(I find it more intuitive to have the smaller value on the left, but don't feel strongly.)_
